webapi/src/main/scala/org/knora/webapi/util/date/CalendarDateUtilV2.scala

```diff
@@ -0,0 +1,543 @@
+  * Represents the era BCE in a Gregorian or Julian calendar date.
+  */
+case object DateEraBCE extends DateEraV2 {
+    override def toString: String = StringFormatter.Era_BCE
+}
+
+/**
+  * Represents the era CE in a Gregorian or Julian calendar date.
+  */
+case object DateEraCE extends DateEraV2 {
+    override def toString: String = StringFormatter.Era_CE
+}
+
+/**
+  * Represents the precision of a date.
+  */
+sealed trait DatePrecisionV2
+
+object DatePrecisionV2 {
+    /**
+      * Parses the name of a date precision.
+      *
+      * @param precisionStr the string to be parsed.
+      * @param errorFun a function that throws an exception. It will be called if the string cannot be parsed.
+      * @return a [[DatePrecisionV2]] representing the date precision.
+      */
+    def parse(precisionStr: String, errorFun: => Nothing): DatePrecisionV2 = {
+        precisionStr match {
+            case StringFormatter.PrecisionDay => DatePrecisionDay
+            case StringFormatter.PrecisionMonth => DatePrecisionMonth
+            case StringFormatter.PrecisionYear => DatePrecisionYear
+            case _ => errorFun
+        }
+    }
+}
+
+/**
+  * Indicates that a date has year precision.
+  */
+case object DatePrecisionYear extends DatePrecisionV2 {
+    override def toString: String = StringFormatter.PrecisionYear
+}
+
+/**
+  * Indicates that a date has month precision.
+  */
+case object DatePrecisionMonth extends DatePrecisionV2 {
+    override def toString: String = StringFormatter.PrecisionMonth
+}
+
+/**
+  * Indicates that a date has day precision.
+  */
+case object DatePrecisionDay extends DatePrecisionV2 {
+    override def toString: String = StringFormatter.PrecisionDay
+}
+
+/**
+  * Represents the name of a calendar.
+  */
+sealed trait CalendarNameV2
+
+object CalendarNameV2 {
+    /**
+      * Parses the name of a calendar.
+      *
+      * @param calendarNameStr a string representing the name of the calendar.
+      * @param errorFun a function that throws an exception. It will be called if the string cannot
+      *                 be parsed.
+      * @return a [[CalendarNameV2]] representing the name of the calendar.
+      */
+    def parse(calendarNameStr: String, errorFun: => Nothing): CalendarNameV2 = {
+        calendarNameStr match {
+            case StringFormatter.CalendarGregorian => CalendarNameGregorian
+            case StringFormatter.CalendarJulian => CalendarNameJulian
+            case _ => errorFun
+        }
+    }
+}
+
+/**
+  * Represents the name of a Gregorian or Julian calendar.
+  */
+sealed trait CalendarNameGregorianOrJulian extends CalendarNameV2
+
+/**
+  * Represents the name of the Gregorian calendar.
+  */
+case object CalendarNameGregorian extends CalendarNameGregorianOrJulian {
+    override def toString: String = StringFormatter.CalendarGregorian
+}
+
+/**
+  * Represents the name of the Julian calendar.
+  */
+case object CalendarNameJulian extends CalendarNameGregorianOrJulian {
```

Comment:
How would that be extended once we support more calendars than Gregorian and Julian?

Reply:
```scala
case object CalendarNameIslamic extends CalendarNameV2
case object CalendarNameHebrew extends CalendarNameV2
```

And so on.

Reply:
got it